### **Data Analytics Task: Analyzing Retail Sales Data**
**Scenario:**
You work as a data analyst for a retail company that operates several stores. Your task is to analyze the last month's sales data for different products across these stores. The dataset includes sales figures (units sold) for three products (Product A, Product B, Product C) across five stores.

**Objective:**
- Calculate the total sales (units sold) for each product across all stores.
- Determine the store with the highest sales for each product.
- Calculate the overall sales across all products and stores.
- Identify the product with the highest sales across all stores.

In [2]:
import numpy as np
np.random.seed(42)
sales_data = np.random.randint(20, 100, size=(5, 3))
sales_data

categories = ["Product A", "Product B", "Product C"]

products_sold = f"Products Sold: Product A - {np.sum(sales_data, axis=0)[0]}, Product B - {
    np.sum(sales_data, axis=0)[1]}, Product C - {np.sum(sales_data, axis=0)[2]}"
highest_sold_per_product = f"Highest Sold Per Product: Product A - Store {np.argmax(sales_data, axis=0)[0]}, Product B - Store {
    np.argmax(sales_data, axis=0)[1]}, Product C - Store {np.argmax(sales_data, axis=0)[2]}"
total_sold = f"Total Sold: {np.sum(sales_data)}"
highest_sold_per_store = f"Highest Sold per Store: {[f'Store {count}: {
    categories[i]}' for count, i in enumerate(np.argmax(sales_data, axis=1))]}"

print(products_sold)
print(highest_sold_per_product)
print(total_sold)
print(highest_sold_per_store)

Products Sold: Product A - 335, Product B - 246, Product C - 249
Highest Sold Per Product: Product A - Store 2, Product B - Store 3, Product C - Store 1
Total Sold: 830
Highest Sold per Store: ['Store 0: Product C', 'Store 1: Product C', 'Store 2: Product A', 'Store 3: Product B', 'Store 4: Product B']


## Interpretation of Data
1. The product with the highest sales is Product A with 335. Product B and Product C almost share the same amount of sales, but they have remarkably lower sales compared to Product A.
    - Product A seems to have a higher demand compared to the other products.
2. Store 2 has sold the most for Product A, Store 3 has sold the most for Product B and Store 1 has sold the most for Product C.
    - In local contexts, Product A is the more popular choice for Store 2.
    - Store 3 customers would prefer Product B.
    - Store 1 customers would prefer Product C.
3. The total products sold for Products A, B and C would total to 830.
4. The most popular product per store would be as follows:

    | Store    | Product |
    | -------- | ------- |
    | 0        | C       |
    | 1        | C       |
    | 2        | A       |
    | 3        | B       |
    | 4        | B       |

## Recommendations:
1. Analyze the demand for Product A, and look at the factors such as store locations, customer demand and product comparisons to find the reason for the Gap between Product A compared to Products B and C.
2. The most popular products per store can also be a factor to track customer demand, so each store can increase their stock for their most popular choices.